modules/apps/frontend-js/frontend-js-a11y-web/src/main/resources/META-INF/resources/A11y.tsx

```diff
@@ -12,185 +12,190 @@
  * details.
  */
 
-import './A11y.scss';
-
-import ClayIcon from '@clayui/icon';
-import ClayList from '@clayui/list';
-import ClayPopover from '@clayui/popover';
-import {ReactPortal} from '@liferay/frontend-js-react-web';
-import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
-
-import useA11y from './useA11y';
+import React, {useCallback, useLayoutEffect, useState} from 'react';
+
+import {ErrorBoundary} from './components/ErrorBoundary';
+import {NotFound} from './components/NotFound';
+import Occurrence from './components/Occurrence';
+import {StackNavigator} from './components/StackNavigator';
+import Violation from './components/Violation';
+import {ViolationPopover} from './components/ViolationPopover';
+import Violations from './components/Violations';
+import useA11y from './hooks/useA11y';
+import {TYPES, useFilterViolations} from './hooks/useFilterViolations';
 
 import type {A11yCheckerOptions} from './A11yChecker';
-import type {Violation as TViolation} from './useA11y';
-
-const rectAttrs: Array<keyof DOMRect> = [
-	'bottom',
-	'height',
-	'left',
-	'right',
-	'top',
-	'width',
-];
-
-interface IRectState {
-	rect: DOMRect | undefined;
-	hasRectChanged: boolean;
-}
 
-const DOMRectStub = {} as DOMRect;
-
-const rectChanged = (a: DOMRect, b: DOMRect = DOMRectStub) =>
-	rectAttrs.some((prop) => a[prop] !== b[prop]);
+type Params = {
+	name?: string;
+	ruleId?: string;
+	target?: string;
+};
 
-const useObserveRect = (
-	callback: (rect: DOMRect | undefined) => void,
-	node: Element | null
-) => {
-	const rafIdRef = useRef<number>();
+export function A11y(props: Omit<A11yCheckerOptions, 'callback'>) {
+	const violations = useA11y(props);
 
-	const run = useCallback(
-		(node: Element, state: IRectState) => {
-			const newRect = node.getBoundingClientRect();
+	const [state, dispatch] = useFilterViolations(violations);
 
-			if (rectChanged(newRect, state.rect)) {
-				state.rect = newRect;
+	const [params, setParams] = useState<Params>();
 
-				callback(state.rect);
-			}
+	const [activePage, setActivePage] = useState(0);
 
-			rafIdRef.current = window.requestAnimationFrame(() =>
-				run(node, state)
-			);
-		},
-		[callback]
-	);
+	const nodes = Object.keys(violations.nodes);
 
-	useEffect(() => {
-		if (node) {
-			run(node, {
-				hasRectChanged: false,
-				rect: undefined,
-			});
+	useLayoutEffect(() => {
+		if (nodes.length > 0) {
```

Comment:
I don't think it's necessary

Reply:
I think it could be both, I like to add `nodes.length > 0` for readability as it's more expressive than just `nodes.length` another option I consider in relation to this is also `Boolean(nodes .length)` for being more expressive as opposed to an implicit check. We already had a talk about this in https://github.com/liferay/liferay-frontend-projects/issues/384 but it was related to JSX.